specs/interop/shared-locbox.md

```diff
@@ -0,0 +1,300 @@
+    require(msg.sender == updater(), "Unauthorized");
+
+    // Add to the dependency set and check it is not already added (`add()` returns false if it already exists)
+    require(_dependencySet.add(_chainId), "Chain already added");
+
+    // Store the system config
+    systemConfigs[_chainId] = _systemConfig;
+
+    // Loop through the dependency set and update the dependency for each chain
+    for (uint256 i; i < _dependencySet.length() - 1; i++) {
+        uint256 currentId = _dependencySet.at(i);
+
+        // Skip recently added chain
+        if (_chainId == currentId) continue;
+
+        systemConfigs[currentId].addDependency(_chainId);
+
+        systemConfigs[_chainId].addDependency(currentId);
+    }
+
+    address portal = _systemConfig.optimismPortal();
+
+    // Authorize the portal on the shared lockbox
+    SHARED_LOCKBOX.authorizePortal(portal);
+
+    emit ChainAdded(_chainId, _systemConfig, portal);
+}
+```
+
+**Getters**
+
+These getter are needed due to using an EnumerableSet.
+
+```solidity
+function dependencySet() external view returns (uint256[] memory) {
+    return dependencySet.values();
+}
+
+function isInDependencySet(uint256 _chainId) public view returns (bool) {
+    return dependencySet.contains(_chainId);
+}
+```
+
+### `OptimismPortal`
+
+This contract will need an upgrade to integrate the `SharedLockbox` and start using the shared ETH liquidity.
+
+**Storage layout**
+
+The storage layout changes include adding the `SharedLockbox` address.
+This address will be immutable because all portals will point to the same lockbox, and this address should not change.
+
+```solidity
+// The Shared Lockbox contract
+address internal immutable SHARED_LOCKBOX;
+```
+
+**Integrating `SharedLockbox`**
+
+The integration with the `SharedLockbox` involves adding extra steps when executing deposit transactions
+or finalizing withdrawal transactions.
+These steps include locking and unlocking ETH without altering other aspects of the current portal implementation.
+To implement this solution, we need the following changes:
+
+- When calling `finalizeWithdrawalTransactionExternalProof` if the token is `ETHER` and `tx.value` is different than zero,
+  then it should call `unlockETH` on the `SharedLockbox` with the `tx.value`.
+- When calling `depositTransaction` if the token is `ETHER` and `msg.value` is different than zero,
+  then it should call `lockETH` on the `SharedLockbox` with the `msg.value`.
+
+### `LiquidityMigrator`
+
+This contract is meant to be used as intermediate step for the liquidity migration.
+It main purpose is to transfer the whole ETH balance from the `OptimismPortal` to the `SharedLockbox`.
+
+A possible migrate function implementation can be:
+
+```solidity
+// The Shared Lockbox contract
+address internal immutable SHARED_LOCKBOX;
+
+function migrateETH() external {
+    SHARED_LOCKBOX.lockETH{ value: address(this).balance }();
+}
+```
+
+### Future Considerations / Additional Notes
```

Comment:
I've never seen questions in a spec. This is common practice for design docs, but for specs you would drope these Qs on the PR, no in the markdown